### DISCUSSION TITLE
Shaping text with feature "init" with the font "Prida01" doesn't seem to work correctly

### DISCUSSION THREAD

**Discussion by LaurenzV** [harfbuzz/harfbuzz#4681] [Q&A] [answered]
(Originally reported at https://github.com/typst/typst/issues/3924, which uses rustybuzz and not harfbuzz, but I discovered that the same issue seems to exist with harfbuzz)

On this site you can find the font: https://www.glukfonts.pl/font.php?font=Prida01

There are two versions offered, a TTF one and an OTF one.

First I tried shaping with the TTF version, which seems to work fine:
`hb-view Prida01.ttf --features init "Hello World" -o out.png`
![out](https://github.com/harfbuzz/harfbuzz/assets/47084093/903a0e71-2957-42df-b219-c5392bf543c5)

While it doesn't work with the OTF version:
`hb-view Prida01.otf --features init "Hello World" -o out.png`
![out](https://github.com/harfbuzz/harfbuzz/assets/47084093/d9fb89a3-7a88-41e5-8bcf-5f31f0f45d76)

At first I thought that this is probably a font issue. But now comes the weird part: I tried subsetting the OTF font with the following command: 
`hb-subset Prida01.otf -o Prida01.otf "Hello World"`

And now the following happens!
`hb-view Prida01.otf --features init "Hello World" -o out.png`
![out](https://github.com/harfbuzz/harfbuzz/assets/47084093/a4398af8-74d9-4e7f-ba84-c27a2f0d41cf)

So I'm not sure what's going on, perhaps it's still a font issue, but it seems weird that somehow subsetting it magically resolves the problem?

(I'm running version 8.4 by the way)

**Comment by khaledhosny** [comment]
The `Prida01.otf` file has a `morx` table while the `Prida01.ttf` does not, so the shaping is different. The font seems to have been generated by FontForge, so the `morx` table is probably automatically generated from the `GSUB` table at build time, most likely by mistake, and it is probably dysfunctional. This is basically a font bug.

**Reply by khaledhosny** [reply, marked as answer]
`hb-subset` drops the `morx` table (it does not support subsetting it), that is why the subset font works as well.

**Reply by LaurenzV** [reply]
Interesting, thanks for looking into it!
